src/lib/coda_intf/staged_ledger_intf.ml

```diff
@@ -299,59 +299,65 @@ module type Transaction_snark_scan_state_generalized_intf = sig
 
   val empty : unit -> t
 
-  val capacity : t -> int
-
   val fill_work_and_enqueue_transactions :
        t
     -> Transaction_with_witness.t list
     -> transaction_snark_work list
-    -> (ledger_proof * transaction list) option Or_error.t
+    -> ((ledger_proof * transaction list) option * t) Or_error.t
 
   val latest_ledger_proof :
     t -> (Ledger_proof_with_sok_message.t * transaction list) option
 
   val free_space : t -> int
 
-  val next_k_jobs : t -> k:int -> Available_job.t list Or_error.t
-
-  val next_jobs : t -> Available_job.t list Or_error.t
-
-  val next_jobs_sequence : t -> Available_job.t Sequence.t Or_error.t
-
   val base_jobs_on_latest_tree : t -> Transaction_with_witness.t list
 
-  val is_valid : t -> bool
-
   val hash : t -> staged_ledger_aux_hash
 
-  val staged_transactions : t -> ledger_undo list
-
-  val all_transactions : t -> transaction list Or_error.t
+  (** All the staged transactions in the reverse order of their application (Latest first)*)
+  val staged_transactions_with_info_rev : t -> ledger_undo list
```

Comment:
Maybe call this staged_undos? Also could maybe have an abstract undos type so the user doesn’t have to know what order the list is in. Like with a function 

Undos.apply : t -> f:(ledger_undo -> unit) -> unit

Which is just List.iter ?